src/lem/constrainer.rs

```diff
@@ -38,7 +38,7 @@ impl<F: LurkField> AllocationManager<F> {
                 let digits = f.hex_digits();
                 let alloc =
                     AllocatedNum::alloc(cs.namespace(|| format!("allocate {digits}")), || Ok(f))
-                        .map_err(|e| anyhow!("{e}: couldn't allocate {digits}"))?;
+                        .with_context(|| format!("couldn't allocate {digits}"))?;
```

Comment:
Does it get rid of the `SynshesisError` in the error message?

Reply:
https://docs.rs/anyhow/1.0.71/anyhow/trait.Context.html#tymethod.with_context